### PR TITLE
movingWindow

### PR DESCRIPTION
**Pull request by Dieterbe**
I took #1874 and
1) fixed the constructor problem
2) rebased on master and adopted the testing best practices

this reveals that output of movingWindow is not canonical. so this will need some more work

### CODE REVIEW

File: expr/func_movingwindow.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+// NewMovingWindowGeneric constructs a generic movingWindow function
+func NewMovingWindowGeneric() GraphiteFunc {
+	return &FuncMovingWindow{
+		fn:      "average",
+		generic: true,
+	}
+}
+
+func (s *FuncMovingWindow) Signature() ([]Arg, []Arg) {
+	if s.fn == "" {
```
Comment (shanson7) [inline expr/func_movingwindow.go:44]:
```suggestion
	if s.generic {
```
?

Comment (Dieterbe) [reply expr/func_movingwindow.go:44]: heh can't believe i missed this :)

Comment (Dieterbe) [reply expr/func_movingwindow.go]: fixed https://github.com/grafana/metrictank/pull/1941/commits/65b9f0c36f77f7c143836160101950c8a01f0811

**Comment by shalstea** [comment]
Any ETA / status on this?

**Review by shanson7 — APPROVED** [review]

**Comment by shanson7** [comment]
> Any ETA / status on this?

Sorry, didn't realize this was in my review queue

**Comment by Dieterbe** [comment]
well, the problem is that the function may seem to work in general, but since the output is not canonical, it will lead to problems when doing further runtime consolidation, combining data with other series, etc.
I think this requires a broader rethink of this particular requirement, as well as cleanup of our testing code (which i added in #1945  and in turn revealed similar problems with a bunch of other functions which needs the same treatment - relaxing of the reqirements or addressing shortcomings). I think this warrants a few days of focused attention and I have that on my list for this month.

**Pull request merged**